crypto/asn1/a_gentm.c

```diff
@@ -157,10 +157,10 @@ int ASN1_GENERALIZEDTIME_print(BIO *bp, const ASN1_GENERALIZEDTIME *tm)
     if (BIO_printf(bp, "%s %2d %02d:%02d:%02d%.*s %d%s",
                    _asn1_mon[M - 1], d, h, m, s, f_len, f, y,
                    (gmt) ? " GMT" : "") <= 0)
-        return (0);
+        return 0;
     else
-        return (1);
+        return 1;
  err:
     BIO_write(bp, "Bad time value", 14);
```

Comment:
Hmmm, the *_print functions in this ‘asn1 time' scope act more like internal functions, most of the following calling patterns:

```
    BIO_printf(bio, "Time stamp: ");
    ASN1_GENERALIZEDTIME_print(bio, a->time);
    BIO_write(bio, "\n", 1);
```
I think the proper behavior for the *caller* is print error message themselves by checking the return value of the asn1_time*_print, e.g.:

```
    BIO_printf(bio, "Time stamp: ");
    if (!ASN1_GENERALIZEDTIME_print(bio, a->time))
        BIO_write("blahblah...");
    BIO_write(bio, "\n", 1);
```
This will simplify the error handling in those 'print' functions.